Arriba/Arriba.Test/CustomColumn.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using Arriba.Model;
+using System.Collections.Generic;
+using Arriba.Model.Expressions;
+using Arriba.Structures;
+using Arriba.Serialization;
+
+namespace Arriba.Test
+{
+    public static class CustomColumnSupport
+    {
+        public static void RegisterCustomColumns()
+        {
+            ColumnFactory.AddColumnCreator("color", (details, columnComponents, initialCapacity) =>
```

Comment:
We might want a different contract. If the custom column gets all of the column components, then it can forget to wrap the custom column in a SortedColumn or an IndexedColumn, for example.